Prometheus Docs (#8)
SteeltoeOSS/Documentation: docs/7-management/13-%2fprometheus.md

### Prometheus

> The Steeltoe Prometheus endpoint configures application metrics collection using the open source [OpenCensus](https://opencensus.io/) project. Similar to the [Metrics Endpoint](12-%2fmetrics), it automatically configures built-in instrumentation of various aspects of the application and exposes the collected metrics in the prometheus format.

Reviewer: is %2f supposed to be in this link?

Author: it looks like it. changing it to / definitely doesn't work

Author: I believe so. the link doesn't work with /

##### Add NuGet References

To use the prometheus endpoint, you need to add a reference to `Steetoe.Management.EndpointCore`. To add this NuGet to your project, add a `PackageReference` resembling the following:

```xml
<ItemGroup>
...
    <PackageReference Include="Steeltoe.Management.EndpointCore" Version= "2.4.2"/>
...
</ItemGroup>
```

or

```powershell
PM>Install-Package  Steeltoe.Management.EndpointCore -Version 2.4.2
```

##### Cloud Foundry

The [Metrics Forwarder for Pivotal Cloud Foundry (PCF)](https://docs.pivotal.io/metrics-forwarder/) is no longer supported on Pivotal Application Service (PAS) v2.5 and later. To emit custom metrics on PAS v2.5 or later, use the Metric Registrar. For more information about enabling and configuring the Metric Registrar, see [Configuring the Metric Registrar](https://docs.pivotal.io/platform/application-service/2-8/metric-registrar/index.html).

To register your endpoint for metrics collection install the metrics-registrar plugin and use it to register your endpoint. 

`cf install-plugin -r CF-Community "metric-registrar"`

`cf register-metrics-endpoint your-dotnet-app /actuator/prometheus`

##### Prometheus Server

> [Prometheus Server](https://prometheus.io/) can be set up to scrape this endpoint by registering it via configuration. For example this prometheus.yml expects steeltoe app running on port 8000 and the actuator management path is the default of /actuator:

Reviewer: ```suggestion
[Prometheus Server](https://prometheus.io/) can be set up to scrape this endpoint by registering your application in the server's configuration. For example this prometheus.yml expects a Steeltoe-enabled app running on port 8000 with the actuator management path at the default of /actuator:
```

```yml
global:
  scrape_interval: 15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'steeltoe-prometheus'
    metrics_path: '/actuator/prometheus'
    scrape_interval: 5s
    static_configs:
      - targets: ['host.docker.internal:8000']
```

> Running Prometheus server with this config will allow you view metrics in the built in UI. Other visualization tools such as [Grafana](https://grafana.com/docs/grafana/latest/features/datasources/prometheus/) can then be configured to use prometheus as a datasource. 

Reviewer: ```suggestion
Running Prometheus server with this configuration will allow you view metrics in the built-in UI. Other visualization tools such as [Grafana](https://grafana.com/docs/grafana/latest/features/datasources/prometheus/) can then be configured to use Prometheus as a datasource. 
```
